exla/lib/exla/defn.ex

```
@@ -300,11 +300,16 @@ defmodule EXLA.Defn do
     EXLA.Op.get_tuple_element(op, index)
   end
 
-  defp to_operator(:dot, [left, axes1, right, axes2], %{type: type}, state) do
+  defp to_operator(:dot, [left, contract_axes1, [], right, contract_axes2, []], %{type: type}, state) do
     precision = state.precision
-    EXLA.Op.dot_general(to_type(left, type), to_type(right, type), {axes1, axes2}, precision)
+    EXLA.Op.dot_general(to_type(left, type), to_type(right, type), {contract_axes1, contract_axes2}, precision)
   end
 
+  defp to_operator(:dot, _, _, _) do
+    raise ArgumentError, "dot operation not implemented for batched tensors"
+  end
+
+
```

Just let it crash on function clause instead. :)

```suggestion
```